src/gamelog.cpp

```diff
@@ -134,11 +134,8 @@ void GamelogReset()
  */
 static char *PrintGrfInfo(char *buf, const char *last, uint grfid, const uint8 *md5sum, const GRFConfig *gc)
 {
-	char txt[40];
-
 	if (md5sum != nullptr) {
-		md5sumToString(txt, lastof(txt), md5sum);
-		buf += seprintf(buf, last, "GRF ID %08X, checksum %s", BSWAP32(grfid), txt);
+		buf += seprintf(buf, last, "GRF ID %08X, checksum %s", BSWAP32(grfid), Md5sumToString(md5sum).c_str());
```

Comment:
Maybe a stupid question, but does the result of `Md5sumToString` live long enough for `seprintf` to be safe to use the `c_str`?

I never know in these statements when the `std::string` is being deleted again, and if that is well defined in C++ to start with.

Reply:
It should be fine, but I've got a better solution... rewrite gamelog's print to use `fmt::format_to` instead, so `c_str()` isn't needed anymore.